### PR TITLE
Join group page

### PR DESCRIPTION
**Pull request by jssunray34**
### Description of change(s), including why:

I worked on creating the Join Group page. Previously, when a user clicked on the Join Group button on the Groups page, a popup would appear prompting them to enter the group code of the group they wanted to join. We decided to remove the popup and instead create a new page for this functionality because dynamically centering the popup on the page became a challenge. Once the user enters the group code and clicks "OK", they must confirm their decision before being redirected to that group's poll page. Additionally, if the user is not logged in when they try to join a group, they are brought to /login.

### Type of change(s)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Any work to still do or other concerns:

- Shows details of the group to join (only if user passes join permission checks)
- Join permissions check - group must be open join by ID, or user must be added to group's allowed to join users list
- Submit request to backend to add user to group
- I have concerns with the way I am checking if a user is logged in (I used localStorage because this is what is used on the login page, but it does not always work)

### Closing issues:

- Makes progress on #290 but does not close

### Screenshots (if frontend change) of before and after:

JoinGroupPage
JoinGroupPageConfirm

### CODE REVIEW

**Comment by jssunray34** [comment]
I created a styled confirmation box (shown below) to be used instead of a default confirmation prompt.
JoinGroupConfirmation

File: PollBuddy-Server/frontend/src/pages/GroupJoin/GroupJoin.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import React, { Component } from "react";
+import "mdbreact/dist/css/mdb.css";
+import { MDBContainer } from "mdbreact";
+
+export default class GroupJoin extends Component {//this class will likely need to call Groups/new and do more with that...
+  componentDidMount(){
+    this.props.updateTitle("Group Join");
```
Comment (hbh7) [inline PollBuddy-Server/frontend/src/pages/GroupJoin/GroupJoin.js:7]: This is probably silly but "Join Group" sounds a lot better to me than "Group Join".

File: PollBuddy-Server/frontend/src/pages/GroupJoin/GroupJoin.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  constructor() {
+    super();
+    //TODO check if they're logged in
+    console.log(localStorage.getItem("loggedIn"));
+    if (localStorage.getItem("loggedIn") == "true") {
```
Comment (hbh7) [inline PollBuddy-Server/frontend/src/pages/GroupJoin/GroupJoin.js:14]: This is fine for now but we may rework that later which could solve the sporadic issues you've been having.

File: PollBuddy-Server/frontend/src/pages/GroupJoin/GroupJoin.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  handleConfirmationResponse(response) {
+    if (response === true) {
+      window.location.replace("/groups/" + this.state.groupCode + "/polls");
+    } else {
+      window.location.replace("/groups");
+    }
+  }
+
+  render() {
+    //TODO check if they're logged in
```
Comment (hbh7) [inline PollBuddy-Server/frontend/src/pages/GroupJoin/GroupJoin.js:43]: Unimportant but just so you know, typical code comment convention is like this: 
```
//This is code that was commented out, usually temporarily
// This is for an information comment. The space is the difference.
// TODO: Is also generally formatted like this
``` 
It definitely doesn't really matter here but I just wanted to point it out as if you go into software development anywhere, that's what you'll see and likely be expected to do. RCOS is all about learning anyways :)

File: PollBuddy-Server/frontend/src/pages/GroupJoin/GroupJoin.js (hunk 1 above)
Comment (hbh7) [reply PollBuddy-Server/frontend/src/pages/GroupJoin/GroupJoin.js:7]: I'll just change it quickly so you don't have to bother with it.

**Review by hbh7 — COMMENTED** [review]
Looks good! There's one tiny thing that I'll just do because it's trivial and not worth asking you to take your time to do. The rest is just general notes, nothing you have to act on. :)

**Review by hbh7 — APPROVED** [review]

**Pull request merged**